src/test/ui/generic-associated-types/issue-86218.rs

```diff
@@ -21,7 +18,8 @@ trait Yay<AdditionalValue> {
 
 impl<'a> Yay<&'a ()> for () {
     type InnerStream<'s> = impl Stream<Item = i32> + 's;
-    fn foo<'s>() -> Self::InnerStream<'s> { todo!() }
+    //^ ERROR does not fulfill the required lifetime
```

Comment:
Is this an intended behaviour? In the original issue it's noted, that with a separate type alias code compiles.

Reply:
nope, but it is consistent until we address https://github.com/rust-lang/rust/issues/96996